seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hadoop/HadoopFileSystemProxy.java

```diff
@@ -273,12 +300,42 @@ private void initializeWithRemoteUserLogin() throws Exception {
                         configuration,
                         hadoopConf.getRemoteUser(),
                         (configuration, userGroupInformation) -> {
-                            final FileSystem fileSystem = FileSystem.get(configuration);
+                            this.userGroupInformation = userGroupInformation;
+                            this.fileSystem = FileSystem.get(configuration);
                             return Pair.of(userGroupInformation, fileSystem);
                         });
         log.info("Create FileSystem success with RemoteUser: {}.", hadoopConf.getRemoteUser());
-        this.userGroupInformation = pair.getKey();
-        this.fileSystem = pair.getValue();
-        this.fileSystem.setWriteChecksum(false);
+        userGroupInformation = pair.getKey();
+        fileSystem = pair.getValue();
+        fileSystem.setWriteChecksum(false);
+    }
+
+    private <T> T execute(PrivilegedExceptionAction<T> action) throws IOException {
+        if (isAuthTypeKerberos) {
+            return doAsPrivileged(action);
+        } else {
+            try {
+                return action.run();
+            } catch (IOException e) {
+                throw e;
+            } catch (SeaTunnelRuntimeException e) {
+                throw new SeaTunnelRuntimeException(e.getSeaTunnelErrorCode(), e.getParams());
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private <T> T doAsPrivileged(PrivilegedExceptionAction<T> action) throws IOException {
+        if (fileSystem == null || userGroupInformation == null) {
+            initialize();
+        }
+
+        try {
+            return userGroupInformation.doAs(action);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
does this necessary?

Reply:
After catching the InterruptedException, thread's interrupt status is cleared, so we need to re-interrupt the thread, for the task threads to continue. CC: @Hisoka-X